src/main/java/de/tum/in/www1/artemis/web/rest/TextClusterResource.java

```diff
@@ -0,0 +1,109 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.TextCluster;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.TextClusterRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.dto.TextClusterStatisticsDTO;
+
+/**
+ * REST controller for managing TextClusterResource.
+ */
+@RestController
+@RequestMapping("/api")
+public class TextClusterResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextClusterResource.class);
+
+    private final TextClusterRepository textClusterRepository;
+
+    private final UserRepository userRepository;
+
+    private final ExerciseRepository exerciseRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public TextClusterResource(TextClusterRepository textClusterRepository, UserRepository userRepository, ExerciseRepository exerciseRepository,
+            AuthorizationCheckService authCheckService) {
+        this.textClusterRepository = textClusterRepository;
+        this.userRepository = userRepository;
+        this.exerciseRepository = exerciseRepository;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * Get /text-exercises/{exerciseId}/cluster-statistics/
+     * <p>
+     * Get text cluster stats
+     * @param exerciseId The id of the text exercise to fetch cluster statistics data from
+     * @return The list of cluster ids adjacent to their respective sizes and automatically graded text blocks
+     */
+    @GetMapping("/text-exercises/{exerciseId}/cluster-statistics")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<TextClusterStatisticsDTO>> getClusterStats(@PathVariable Long exerciseId) {
```

Comment:
Having just reviewed the resource refactoring PR's of @PatZintel, I got the impression that we are in the progress of removing the starting `/` in the routes. However, there are no guidelines on that. Especially as question to Patrick, should we always use routes without the `/`?

Reply:
Was not aware of this 🤔 . Searching for `@GetMapping("/` , I get 100+ matches already with the `/` in the repo. But can always change it if this is the new normal 👍

Reply:
@MaisiKoleni actually, there are guidelines on this ^^' 
![image](https://user-images.githubusercontent.com/80530171/131336265-c9c0bfeb-d666-43b9-8995-a111f3338fc7.png)
this is exempt from our server coding and design guidelines, a section about the RestControllers. And I am aware that there are still a lot of matches when searching for that, but that's why i am doing all those refactorings xD